Fails to build on FreeBSD

Regressed by 46ee0707901e. Defining `_POSIX_C_SOURCE` to an empty value is likely illegal. Not to be confused with `_POSIX_SOURCE` which can be empty and implies the first version aka `-D_POSIX_C_SOURCE=198808`. 
```
$ cc --version
FreeBSD clang version 14.0.5 (https://github.com/llvm/llvm-project.git llvmorg-14.0.5-0-gc12386ae247c)
Target: x86_64-unknown-freebsd14.0
Thread model: posix
InstalledDir: /usr/bin
```

```c
$ meson setup /tmp/labwc_build
$ meson compile -C /tmp/labwc_build
FAILED: labwc.p/src_action.c.o
cc -Ilabwc.p -I. -I../../path/to/labwc -Iinclude -Iinclude -Isubprojects/wlroots/include -Isubprojects/wlroots/include -I/usr/local/include -I/usr/local/include/libepoll-shim -I/usr/local/include/libdrm -I/usr/local/include/pixman-1 -I/usr/local/include/libxml2 -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include/cairo -I/usr/local/include/freetype2 -I/usr/local/include/libpng16 -I/usr/local/include/harfbuzz -I/usr/local/include/pango-1.0 -I/usr/local/include/fribidi -fcolor-diagnostics -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wundef '-DLABWC_VERSION="0.5.0-224-g1eab4b5"' '-DGETTEXT_PACKAGE="labwc"' '-DLOCALEDIR="/usr/local/share/locale"' -pthread -D_THREAD_SAFE -MD -MQ labwc.p/src_action.c.o -MF labwc.p/src_action.c.o.d -o labwc.p/src_action.c.o -c src/action.c
In file included from src/action.c:3:
In file included from /usr/include/strings.h:34:
/usr/include/sys/cdefs.h:658:49: error: invalid token at start of a preprocessor expression
#if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE == 1
                                                ^
/usr/include/sys/cdefs.h:664:49: error: invalid token at start of a preprocessor expression
#if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE == 2
                                                ^
/usr/include/sys/cdefs.h:694:21: error: invalid token at start of a preprocessor expression
#if _POSIX_C_SOURCE >= 200809
                    ^
/usr/include/sys/cdefs.h:697:23: error: invalid token at start of a preprocessor expression
#elif _POSIX_C_SOURCE >= 200112
                      ^
/usr/include/sys/cdefs.h:700:23: error: invalid token at start of a preprocessor expression
#elif _POSIX_C_SOURCE >= 199506
                      ^
/usr/include/sys/cdefs.h:703:23: error: invalid token at start of a preprocessor expression
#elif _POSIX_C_SOURCE >= 199309
                      ^
/usr/include/sys/cdefs.h:706:23: error: invalid token at start of a preprocessor expression
#elif _POSIX_C_SOURCE >= 199209
                      ^
/usr/include/sys/cdefs.h:709:23: error: invalid token at start of a preprocessor expression
#elif _POSIX_C_SOURCE >= 199009
                      ^
8 errors generated.
```
See also https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_02_01_01

--- a/src/action.c
+++ b/src/action.c
@@ -1,7 +1,9 @@
 // SPDX-License-Identifier: GPL-2.0-only
-#include <strings.h>
-#include <wlr/util/log.h>
+#define _POSIX_C_SOURCE 200809L
 #include <signal.h>
+#include <strings.h>
+#include <unistd.h>
+#include <wlr/util/log.h>
 #include "common/spawn.h"
 #include "common/zfree.h"
 #include "labwc.h"

